### ISSUE TITLE
Sometimes swiping gesture refreshes weather data

### ISSUE THREAD

**Issue by efb4f5ff-1298-471a-8973-3d47447115dc** [martykan/forecastie#592]
When i swipe to the right, it doesnt let me. Instead the guesture gets blocked by the refresh action.

https://user-images.githubusercontent.com/73130443/122406876-b717a900-cf70-11eb-8acd-5d623805809d.mp4

**Comment by robinpaulson** [comment]
Huh, interesting yes. I guess some people swipe slightly different to others. Maybe we change the "swipe-sensitive" area to only the top half of the screen, the part with the "weather now" information?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Just for clarification, i tried swiping in allot of ways i didn't even know about.

i get the same result every time .

**Comment by robinpaulson** [comment]
@andrewmattie Given you originally programmed the "swipe to refresh" PR, would you be able to take a look at this? How difficult would it be to only allow the swipe action for the top half of the screen? Do you see any potential problems if we do that?

Thanks

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Just to clarify this also happens on the top half of the screen

https://user-images.githubusercontent.com/73130443/131841338-65d1f4c7-30c3-40fd-ad50-70f33007f2fb.mp4

**Comment by Zocker1999NET** [comment]
I do not know how to implement it, but other apps seem to allow both swiping gestures everywhere (not only at the top of the screen) but decrease their sensitivity for both. E.g. if you move your finger more on the horizontal than on the vertical axis, then the app will switch the tab accordingly. On the opposite, if you move your finger more on the vertical than on the horizontal axis, then the app will refresh (if it was from top to bottom) but will not change the current tab. Sometimes in those apps, you even see both animations at the same time (the "reloading circle" at the top becomes visible and the tab moves) but it aborts one of both actions/animations as you continue to swipe.

For Forecastie, I think it would be better to implement it in that way, if possible and easy, so you can also refresh from the lower half of the screen. As I see it, the current problem seems to be that if you begin to swipe only a single pixel from top to bottom, then the "tab change" animation will be aborted immediately as the "reloading" animation seems to be super more important.

**Comment by robinpaulson** [comment]
Huh, very interesting. It sounds like the sort of thing which is set via one or two variables associated with the procedure call. Should be easy to find and modify.

What do you think about doing some research around this?

**Comment by robinpaulson** [comment]
This is the PR to add the "swipe to refresh" feature.

**Comment by oiao** [comment]
Bump, as I also have that issue. Swiping between tabs rarely works in my case. An option to disable pull to refresh in the settings might be a good solution, since there already is a refresh button

**Comment by robinpaulson** [comment]
I think the settings page is already vastly overloaded, the amount of configurability we've built in is bordering on overwhelming. I'm not in favour of adding more things, I think we should be taking some away. At the moment it feels like Forecastie is an app for clicking buttons and selecting list items; getting the weather forecast appears a distant second.

**Comment by oiao** [comment]
Right, I was just offering an easy temporary solution to that issue if fixing it properly requires more time
